secboot/secboot_hooks.go

```diff
@@ -174,9 +166,25 @@ func unlockVolumeUsingSealedKeyFDERevealKeyV2(sealedEncryptionKeyFile, sourceDev
 		return res, xerrors.Errorf(fmt, err)
 	}
 
+	model, err := opts.WhichModel()
+	if err != nil {
+		return res, fmt.Errorf("cannot retrieve which model to unlock for: %v", err)
+	}
+
 	// the output of fde-reveal-key is the unsealed key
 	options := activateVolOpts(opts.AllowRecoveryKey)
-	modChecker, err := sbActivateVolumeWithKeyData(mapperName, sourceDevice, keyData, options)
+	// FIXME: consider setting it in activateVolOpts if we keep
+	// this function separate from the tpm one for key data in
+	// files. Otherwise we should only set it if we provide key
+	// data generation 1.
+	options.Model = model
+
+	authRequestor, err := newAuthRequestor()
+	if err != nil {
+		return res, fmt.Errorf("internal error: cannot build an auth requestor: %v", err)
+	}
+
+	err = sbActivateVolumeWithKeyData(mapperName, sourceDevice, authRequestor, sb.Argon2iKDF(), options, keyData)
```

Comment:
Does it work if you pass `nil` instead of `sb.Argon2iKDF`? The intention here is that this API would always be executed remotely in a short-lived helper process because otherwise there's a need to force a GC between invocations. As there's technically no passphrase support right now, nothing should call this anyway.

Reply:
This parameter is removed in the next queued up PR. But I can change that and see if the tests are still working

Reply:
Unlocking failed with:
```
[    4.522158] snap-bootstrap[195]: 2024/08/22 12:44:12.816188 main.go:63: execution error: cannot unlock encrypted partition: nil kdf
```